### ISSUE TITLE
Allow setting default time zone for content date strings.

### ISSUE THREAD

**Issue by pbannister** [11ty/eleventy#2968]
This is not a new issue, as is listed in "Common Pitfalls". When I write in front-matter:
```
---
date: 2004-07-03
---
```
What I meant is 2004-07-03 in California, not in Greenwich. 
The default in Eleventy is to assume UTC, which is perfectly fine ... as a default.
I fully expect to have to tell Eleventy this in configuration, as behavior local to my website.

... soon as I can figure out where to over-ride. :)
(``TemplateContent.read()`` looks promising, maybe.)

To my mind, the public work-arounds are backwards. 
I do not want to over-ride the *display* of the date, I want to over-ride the front-matter date *parsing*.

In short, I want to teach Eleventy to "Do What I Mean".

**Comment by zachleat** [comment]
This is part of the YAML specification, unfortunately, and not Eleventy.

Date strings might be open to configuration, though—e.g.

```
---
date: "2004-07-03"
---
```

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by pbannister** [comment]
Going to hit this once, then drop, and solve off-line.

You are not in any way obligated to mindless conformity with the YAML specification. YAML calls for datetime to be UTC, and that in an entirely reasonable default. You are using YAML within the context of your application. Most of the folk that use your application are within a single timezone, and that timezone is not UTC.

Your default, lacking any other information, should follow the YAML default.

But if I add information - if I tell you my date-times are local, then now you know better.

You are not a computer program exchanging information with another computer program. You are a human, writing software for other humans.

If I were a multi-national corporation, my posters could be from many different time-zones, so staying with UTC as a default might make sense.

If I am not a multi-national corporation, as very likely true for most folk, then I am local to a single zone. I need to state to you my locality (so not a default), but will express time in my context.

When I write a date, _I mean exactly what I wrote_, from my context.

**Comment by zachleat** [comment]
This is possible via https://www.11ty.dev/docs/dates/#configuration-api-for-custom-date-parsing (shipped in v3.0.0) but we’ll also have a bit more sugar around setting a default default time zone in #3668
